src/js/components/DataTable/Resizer.js

```diff
@@ -12,55 +12,61 @@ const ResizerBox = styled(Box)`
   cursor: col-resize;
 `;
 
-class Resizer extends Component {
-  state = {};
+const Resizer = ({ onResize, property, theme }) => {
+  const [state , setState] = useState({
+    start: undefined,
+    width: undefined,
+    event: undefined,
+  });
+  const ref = useRef();
 
-  ref = React.createRef();
+  const onMouseMove = event => {
+    // We determined 12 empirically as being wide enough to hit but
+    // not too wide to cause false hits.
+    const nextWidth = Math.max(12, state.width + (event.clientX - state.start));
+    onResize(property)(nextWidth);
+  };
 
-  onMouseDown = event => {
-    if (this.ref.current) {
-      let element = this.ref.current;
+  const onMouseUp = () => {
+    document.removeEventListener('mouseup', onMouseUp);
+    document.removeEventListener('mousemove', onMouseMove);
+    setState({ ...state, start: undefined, width: undefined });
+  };
+
+  const onMouseDown = event => {
+    if (ref.current) {
+      let element = ref.current;
       // find TH parent
       while (element && element.nodeName !== 'TH') element = element.parentNode;
       const rect = element.getBoundingClientRect();
-      this.setState({ start: event.clientX, width: rect.width }, () => {
-        document.addEventListener('mousemove', this.onMouseMove);
-        document.addEventListener('mouseup', this.onMouseUp);
+      setState({
+        start: event.clientX,
+        width: rect.width,
+        event: 'onMouseDown',
       });
     }
   };
 
-  onMouseMove = event => {
-    const { onResize, property } = this.props;
-    const { start, width } = this.state;
-    // We determined 12 empirically as being wide enough to hit but
-    // not too wide to cause false hits.
-    const nextWidth = Math.max(12, width + (event.clientX - start));
-    onResize(property)(nextWidth);
-  };
-
-  onMouseUp = () => {
-    document.removeEventListener('mouseup', this.onMouseUp);
-    document.removeEventListener('mousemove', this.onMouseMove);
-    this.setState({ start: undefined, width: undefined });
-  };
+  useEffect(() => {
+    if (state.event === 'onMouseDown') {
+      document.addEventListener('mousemove', onMouseMove);
+      document.addEventListener('mouseup', onMouseUp);
```

Comment:
Its a good habit to pass the dependencies to the useEffect array,
```js
[state,onMouseMove,onMouseUp]
```
also, its a good idea to use useCallback with onMouseMove and onMouseUp

Reply:
Thank you so much, fixed!